rust/connlib/tunnel/src/io.rs

```diff
@@ -186,9 +212,65 @@ pub enum DnsQueryError {
     TooManyQueries,
 }
 
+/// Exactly the same as the [TokioRuntimeProvider](hickory_resolver::name_server::TokioRuntimeProvider) but sockets are protected when created
```

Comment:
Technically, we don't know that here, right? So maybe say something like:

```suggestion
/// Identical to [`TokioRuntimeProvider`](hickory_resolver::name_server::TokioRuntimeProvider) but using our own [`SocketFactory`].
```